test/unit/tech/flash.test.js

```diff
@@ -197,6 +197,33 @@ test('play after ended seeks to the beginning', function() {
   equal(seeks[0], 0, 'seeked to the beginning');
 });
 
+test('duration returns NaN, Infinity or duration according to the HTML standard', function() {
+  let duration = Flash.prototype.duration;
+  let mockedDuration = -1;
+  let mockedReadyState = 0;
+  let result;
+  let mockFlash = {
+    el_: {
+      vjs_getProperty() {
+        return mockedDuration;
+      }
+    },
+    readyState: function() {
+      return mockedReadyState;
+    }
+  };
+  result = duration.call(mockFlash);
+  ok(Number.isNaN(result), 'duration returns NaN when readyState equals 0');
```

Comment:
Looks like this and `Number.isFinite` below the two things causing issues. I'll fix it by including `es6-shim` in the test builds.